04.ls/ls.rb

```diff
@@ -0,0 +1,31 @@
+#! /usr/bin/env ruby
+# frozen_string_literal: true
+
+COLUMNS = 3
+COLUMNS_DISTANCE = 5
+
+def main(file_names)
+  return exit if file_names.empty?
+  longest_file_name = file_names.max_by(&:size)
+  width = longest_file_name.size + COLUMNS_DISTANCE
```

Comment:
上の画像はオリジナルのlsと列幅が揃っていますが、下の方が揃ってないですね。

<img width="704" height="769" alt="Image" src="https://github.com/user-attachments/assets/c1829454-f2a9-44c4-afc2-41dcebace530" />

何か規則性があってそれを再現できそうならがんばって再現してほしいですが、難しそうなら単純に

```suggestion
  width = longest_file_name.size + 1
```

だけでもOKです。
（`COLUMNS_DISTANCE = 5`だと、あるときは揃うが、あるときは揃わない中途半端な状態になるので）

Reply:
規則性は見つけられませんでいたが、出力されるファイル名の後ろにタブを挿入するとオリジナルのlsと列幅が揃いました。
ただ、オリジナルのlsコマンドでは列の間はタブしかなく、最終列の後ろには何も出力されていませんでした。
こちらを再現するのは難しそうです💦
<img width="816" height="465" alt="_marukokazuki___Desktop_test__•_Untitled-1" src="https://github.com/user-attachments/assets/0a3717f8-bdbf-416a-a8b7-f67bad214c8b" />

Reply:
> ただ、オリジナルのlsコマンドでは列の間はタブしかなく、最終列の後ろには何も出力されていませんでした。こちらを再現するのは難しそうです💦

いったん配列に書く文字列を放り込んでタブ文字でjoinすると、再現できるかも？